compiler-plugin-test/src/test/kotlin/it/unibo/collektive/test/LoopWithoutAlignTest.kt

```diff
@@ -0,0 +1,62 @@
+package it.unibo.collektive.test
+
+import io.kotest.core.spec.style.FreeSpec
+import it.unibo.collektive.test.util.CompileUtils
+import it.unibo.collektive.test.util.CompileUtils.ProgramTemplates
+import it.unibo.collektive.test.util.CompileUtils.noWarning
+import it.unibo.collektive.test.util.CompileUtils.warning
+import org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi
+
+@OptIn(ExperimentalCompilerApi::class)
+class LoopWithoutAlignTest : FreeSpec({
+    "When being inside a loop in an Aggregate function" - {
+        val testingProgramTemplate = CompileUtils.testingProgramFromTemplate(
+            ProgramTemplates.SINGLE_AGGREGATE_IN_A_LOOP,
+        )
+        listOf(
+            "exampleAggregate" to "exampleAggregate()",
+            "neighboring" to "neighboring(0)",
+        ).forEach { (functionName, functionCall) ->
+            "using $functionName without a specific alignedOn" - {
+                "should produce a warning" - {
+                    val testingProgram = testingProgramTemplate
+                        .put("aggregate", functionCall)
+                    testingProgram shouldCompileWith warning(EXPECTED_WARNING_MESSAGE.format(functionName))
+                }
+            }
+            "using $functionName wrapped in a specific alignedOn" - {
+                val testingProgram = testingProgramTemplate
+                    .put("beforeAggregate", "alignedOn(pivot(localId)) {")
```

Comment:
I'm pretty sure that creating a pivot based on `localId` can be problematic in real code: since all the devices have different IDs, no one would align at the same path (pivot) since it is built based on the ID.

Since here we are testing whether the warning is raised or not, but since the tests act (among the many) as a specification, I would not make the pivot dependent on the `localId`, what do you think @angelacorte @DanySK ?

Reply:
For testing purposes, I don't really see the problem

Reply:
I would have preferred a static string instead, but that's okay.
Mark as resolved.